controller.py

```diff
@@ -367,8 +368,9 @@ def _make_request(url, on_404, on_timeout, on_exception, allow_401=False):
             opds_url = auth_document.root['href']
         else:
             failure_detail = _("The OPDS authentication document is missing a 'start' link to the root OPDS feed.")
-        if auth_document.id != auth_response.url:
-            failure_detail = _("The OPDS authentication document's id (%(id)s) doesn't match its url (%(url)s).", id=auth_document.id, url=auth_response.url)
+
+        if auth_document.id != auth_response.final_url:
+            failure_detail = _("The OPDS authentication document's id (%(id)s) doesn't match its url (%(url)s).", id=auth_document.id, url=auth_response.final_url)
```

Comment:
This handles the case where there's a redirect between the URL provided in the registry submission and the final URL of the authentication document. This required a number of minor changes to the tests, but setting up a redirect is another way to change which URL you're using for a circulation manager.